db/version_set.cc

```diff
@@ -4729,25 +4731,32 @@ class ManifestPicker {
 
  private:
   const std::string& dbname_;
-  FileSystem* const fs_;
   // MANIFEST file names(s)
   std::vector<std::string> manifest_files_;
   std::vector<std::string>::const_iterator manifest_file_iter_;
+  const std::vector<std::string>& dbname_children_;
   Status status_;
 };
 
-ManifestPicker::ManifestPicker(const std::string& dbname, FileSystem* fs)
-    : dbname_(dbname), fs_(fs) {}
+ManifestPicker::ManifestPicker(const std::string& dbname,
+                               const std::vector<std::string>& dbname_children,
+                               const Status& dbname_children_s)
+    : dbname_(dbname),
+      dbname_children_(dbname_children),
+      status_(dbname_children_s) {}
```

Comment:
Does it need to be named `dbname_children_s` which has little relationship with class ManifestPicker itself?

Also, consider to read the naming section of Google C++ Style: https://google.github.io/styleguide/cppguide.html#Naming and try to think of a better name.

Reply:
Thanks for the advice. I'll for sure use a better name if we decide to keep this piece of code in the end.

The biggest issue is that the result of `getChildren()` in `DBImpl::Recover()`, which is passed in here, is supposed to be identical to the result of `fs_->getChildren()` in the original code. But it wasn't.

So I'm going to revert this change in my next commit, and hope we can roll it back once we figured out how to use existing `getChildren()` results from `Recover()`.

Reply:
The reason for the difference you mentioned above is because `NewDB()` may be called if RocksDB decides to create a new database. In `NewDB()`, a `MANIFEST-000001` will be created. You may consider modifying this function so that the MANIFEST file name can be returned.

Reply:
So this means that we can simply use the (known) manifest file name returned by (updated) `NewDB()` and create a result, instead of running `GetChildren()` again?

Reply:
Yes if `NewDB()` is called.